client/components/duplicates/index.js

```diff
@@ -0,0 +1,107 @@
+import { store, State } from '../../store';
+import { debug as dbg, translate as $t } from '../../helpers';
+import SimilarityPairComponent from './item';
+
+function debug(text) {
+    return dbg(`Similarity Component - ${text}`);
+}
+
+// Algorithm
+
+function findRedundantPairs(operations, duplicateThreshold) {
+    let before = Date.now();
+    debug('Running findRedundantPairs algorithm...');
+    debug(`Input: ${operations.length} operations`);
+    let similar = [];
+
+    // duplicateThreshold is in hours
+    let threshold = duplicateThreshold * 60 * 60 * 1000;
+    debug(`Threshold: ${threshold}`);
+
+    // O(n log n)
+    let sorted = operations.slice().sort((a, b) => a.amount - b.amount);
+    for (let i = 0; i < operations.length; ++i) {
+        let op = sorted[i];
+        let j = i + 1;
+        while (j < operations.length) {
+            let next = sorted[j];
+            if (next.amount !== op.amount)
+                break;
+            let datediff = Math.abs(+op.date - +next.date);
+            // Two operations are duplicates if they were not imported at the same date.
+            if (datediff <= threshold && +op.dateImport !== +next.dateImport)
+                similar.push([op, next]);
+            j += 1;
+        }
+    }
+
+    debug(`${similar.length} pairs of similar operations found`);
+    debug(`findRedundantPairs took ${Date.now() - before}ms.`);
+    // The duplicates are sorted from last imported to first imported
+    similar.sort((a, b) => Math.max(b[0].dateImport, b[1].dateImport) -
+        Math.max(a[0].dateImport, a[1].dateImport)
```

Comment:
This is a bit ugly to read. Could each call to Math.max be on its own line, with the minus at the end of the first line?